### ISSUE TITLE
Site Hub: Make the View Site link always visible

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#57421]
### Description

In the Site Editor, at the top of the navigation panel, the 'site hub' shows a View Site link only when the site hub itself is hovered or the link is focused. Screenshot:

![Screenshot 2023-12-28 at 10 32 29](https://github.com/WordPress/gutenberg/assets/1682452/5daf7b63-4262-4554-b509-1cbfd9295310)

While I do realize the intent to make the site hub 'cleaner', I'm not sure _hiding_ an important functionality is ideal. To me, the only effect is that this link is less discoverable.
- If this link is important enough, it should always be visible. If it's not important, it should be removed.
- 'cleanliness' of the UI is a subjective perception that should never have priority on functionality.
- On mobile there is no hover state, and users rarely use keyboards connected to a mobile device so this link is basically not available to mobile devices.
- It's (almost?) a unique case in the whole editor UI where an interactive control is only revealed on hover / focus. As such, it's inconsistent and provides an unexpected interaction.

### Step-by-step reproduction instructions

- Go to the Site Editor.
- Hover your pointing device on the top part of the navigation panel.
- Observe the View Site link becomes visible.
- Move your pointing device away from the  top part of the navigation panel.
- Observe the View Site link is now hidden.
- Use the keyboard to navigate the UI by pressing the Tab key.
- Observe the View Site link becomes visible on focus.
- Tab away from the View Site link.
- Observe the View Site link is now hidden.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Adding the accessibility label, as the cognitive load an unexpected interaction are an accessibility issue IMHO.

**Comment by carolinan** [comment]
I agree with this assessment.
There are additional comments from members of the accessibility team on the original pull request:
https://github.com/WordPress/gutenberg/pull/50420#issuecomment-1642528823
